### ISSUE TITLE
Can't open certain pre-existing files after updates from 8.0.5 through 8.2.2

### ISSUE THREAD

**Issue by Innenseide** [owncloud/core#21720]
Sorry if this is not 100% accurately filled, I'm in a hurry now after playing around for hours. Posting this issue as recommended here: https://forum.owncloud.org/viewtopic.php?f=36&t=32538&p=105222#p105222

### Steps to reproduce
1. Upgrade fully running 8.0.5 installation through 8.0.10, 8.1.5 and 8.2.2
2. Verified files are accessible for every upgrade step by looking at some pictures, opening some plain text files

### Expected behaviour
I should be able to open and use all files.

### Actual behaviour
I can't open/download/use certain files. I can't really make out a pattern. All .pdf files I found are affected, some plain text files with "unusual" (non .txt, like .log, .cfg) or no extension at all, some .zip files (but not all).

PDF.js error:
PDF.js Version 1.0.712 (build: 6969ed4)
Nachricht: Unexpected server response (0) while retrieving PDF "https://url.abc/index.php/apps/files/ajax/download.php?dir=%2F&files=pdf-file.pdf".

Downloading through web browsers or webdav file managers will fail, telling me that the source file is not readable (translated from German).
A similar error is thrown by the sync client (2.0.1, ubuntu) once it progresses to checking one of the malfunctioning files (don't have the logfile right now, sorry, will deliver when home), completely halting the syncronisation process (red x symbol).

Plain text files become usable when renamed to .txt, unusable again when reverted to the previous extension. When I make a change to the file while it's called .txt, it will remain useable once given it's old extension.
pdf files can be renamed to .txt to download them. Locally renaming them back to .pdf will yield a perfectly fine .pdf file. Re-uploading this file through the browser will fix the issue for this file.

I'm still playing around, will add info as I find it.

### Server configuration

running on share hoster all-inkl.com

**Operating system**: Linux 3.2.0-95-generic Ubuntu

**Web server:** apache2

**Database:** mysql

**PHP version:** PHP Version 5.4.45-nmm1

**ownCloud version:** 8.2.2

**Updated from an older ownCloud or fresh install:** Updated since something oC 5, running fine until 8.0.5

**List of activated apps:**

activity
encryption
deleted files
first run wizard
gallery
mail template editor
pdf viewer
share files
text editor
updater
versions
video viewer
calendar
contacts
news
tasks
ownnote
rainloop

**The content of config/config.php:**

$CONFIG = array (
'instanceid' => '',
'passwordsalt' => '',
'trusted_domains' =>
array (
0 => 'my.domain',
),
'datadirectory' => '/www/htdocs/.../owncloud/data',
'dbtype' => 'mysql',
'version' => '8.2.2.2',
'dbname' => '...',
'dbhost' => 'localhost',
'dbtableprefix' => 'oc_',
'dbuser' => '...',
'dbpassword' => '...',
'installed' => true,
'appcodechecker' => false,
'loglevel' => 1,
'log_rotate_size' => 10485760,
'maintenance' => false,
'theme' => '',
'secret' => '...',
'forceSSLforSubdomains' => true,
'trashbin_retention_obligation' => 'auto',
'forcessl' => true,
'defaultapp' => 'calendar',
);

**Are you using external storage, if yes which one:** no

**Are you using encryption:** yes

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox 43

**Operating system:** Windows 7

#### Web server error log
---

#### ownCloud log (data/owncloud.log)
nothing related here, sorry.

#### Browser log
---

**Comment by karlitschek** [comment]
This looks like some kind of mime type problem on your server setup. I recommend that you look at support option at owncloud.org/support. I don't think this is an ownCloud bug. It looks more like an environment or webserver setting bug. Thanks

**Comment by LukasReschke** [comment]
Can you by any chance try with another browser than Firefox? Reminds me a bit of https://github.com/owncloud/enterprise/issues/943

**Comment by LukasReschke** [comment]
So can you view, open and download the file using Chrome for example?

**Comment by PVince81** [comment]
Possibly related to wrong file sizes due to encryption migration issue, see https://github.com/owncloud/core/issues/18225

Pick a broken file, then do: `select * from oc_filecache where path like '%thefilename%'`;
If "unencrypted_size" is equal to "size", then this means the unencrypted size is wrong and it will cause problems on download.

CC @schiesbn

**Comment by Innenseide** [comment]
Thanks for reopening and taking the time.

The issue is just the same on Chrome.

After checking for what @PVince81 said I think I may have messed up this one beyond repair though... unencrypted_size is 0 on **every** entry in oc_filecache, even for files I uploaded just now. Also, encrypted is 0 on every entry except for those files I already tinkered with or uploaded right now.
Then again, the vast majority of files is working 100% fine, as far as I can see.

**Comment by Innenseide** [comment]
I know what happened now. **tl;dr:** truncate oc_filecache is a bad idea.

It seems that with encrypted files, the size value in oc_filecache is created too large (4th visible column):
![wrong](https://cloud.githubusercontent.com/assets/13463550/12415842/2de116b4-be9e-11e5-8998-aebc5e552ab3.png)

This is the result of running the experimental app from #18225:
![right](https://cloud.githubusercontent.com/assets/13463550/12415841/2de00c38-be9e-11e5-9da8-9e665e97e2c2.png)

I had to modify the app, because it identified encrypted files by checking if they start with `HBEGIN:oc_encryption_module:OC_DEFAULT_MODULE:cipher:AES-256-CFB:HEND` - however my encrypted files started with `HBEGIN:cipher:AES-256-CFB:HEND` - I exchanged those patterns in the script so it would work on my files, and it does. Is there anything wrong with that?

Anyways, after the correction, the files work flawlessly again.
I noticed that the encrypted value is 0 on **every** file and the app corrects that, but that doesn't seem to matter; other encrypted files work with 0 or 1.

I can reproduce the problem by truncating oc_filecache and triggering its repopulation by accessing the files app. After that, pdf files don't work, with the size being too large (picture 1). This also affects files I uploaded after upgrading to 8.2.2! After running the 18225 app over them the size is decrased (picture 2), and the files are working fine again.

I know I'm not supposed to just truncate tables at my pleasure - but shouldn't the size values be created right without running a repair app?

(I started truncating oc_filecache in the first place because my "last changed" values were all over the place after the upgrades for whatever reason, and some guys I found via Google said truncate oc_filecache will take care of that).
